SELinux is not required for libvirt, but if it's enabled, libvirt will
fail if it doesn't mount /selinux.  Make the selinux mounts optional.

Index: libvirt-1.0.6/src/lxc/lxc_container.c
===================================================================
--- libvirt-1.0.6.orig/src/lxc/lxc_container.c
+++ libvirt-1.0.6/src/lxc/lxc_container.c
@@ -667,6 +667,7 @@ static int lxcContainerMountBasicFS(char
         const char *type;
         const char *opts;
         int mflags;
+	int optional
     } mnts[] = {
         /* When we want to make a bind mount readonly, for unknown reasons,
          * it is currently necessary to bind it once, and then remount the
@@ -680,8 +681,8 @@ static int lxcContainerMountBasicFS(char
         { "sysfs", "/sys", "sysfs", NULL, MS_NOSUID|MS_NOEXEC|MS_NODEV },
         { "sysfs", "/sys", "sysfs", NULL, MS_BIND|MS_REMOUNT|MS_RDONLY },
 #if WITH_SELINUX
-        { SELINUX_MOUNT, SELINUX_MOUNT, "selinuxfs", NULL, MS_NOSUID|MS_NOEXEC|MS_NODEV },
-        { SELINUX_MOUNT, SELINUX_MOUNT, NULL, NULL, MS_BIND|MS_REMOUNT|MS_RDONLY },
+        { SELINUX_MOUNT, SELINUX_MOUNT, "selinuxfs", NULL, MS_NOSUID|MS_NOEXEC|MS_NODEV, .optional=1 },
+        { SELINUX_MOUNT, SELINUX_MOUNT, NULL, NULL, MS_BIND|MS_REMOUNT|MS_RDONLY, .optional=1 },
 #endif
     };
     int i, rc = -1;
@@ -722,7 +723,8 @@ static int lxcContainerMountBasicFS(char
                                  _("Failed to mount %s on %s type %s flags=%x opts=%s"),
                                  srcpath, mnts[i].dst, NULLSTR(mnts[i].type),
                                  mnts[i].mflags, NULLSTR(mnts[i].opts));
-            goto cleanup;
+	    if (!mnts[i].optional)
+		goto cleanup;
         }
     }
 
